### DISCUSSION TITLE
Vector similarity search speed

### DISCUSSION THREAD

**Discussion by Kencho5** [milvus-io/milvus#15383] [Q&A and General discussion]
hello,

i'm extracting feature vectors from images and each vector size is 2048. after that i'm inserting vectors into the Milvus database. i will have around 170 million image vectors inside Milvus and i was wondering will it still take under a second to search similar vectors from Milvus?

thanks.

**Comment by yhmo** [comment]
Dimension 2048 is too high, it can't take under a second to search. Typically we use 128-dimensional vectors to test.

**Reply by Kencho5** [reply]
how many seconds would it take if you could estimate on a 8 core machine let's say and if i use 128 dimensional vectors then i won't be able to match the images, no? it'd be very inaccurate

**Reply by yhmo** [reply]
10M vectors with 128 dimensional
IVF_FLAT index nlist = 4096
index_file_size = 1024

My machine CPU 3.20GHz 12 cores.
Milvus v1.1.1

Search parameter: nq=1, nprobe=16

Time cost for each search request is about **3** seconds;

**Comment by Kencho5** [comment]
what if i use 512 dimensional vectors? will it still be slow?

**Reply by yhmo** [reply]
For 512 dimensional vectors, you need a powerful machine, and with IVF_PQ or HNSW index, you may achieve "a second to search".

**Reply by Kencho5** [reply]
don’t know if this is the method to choose for my project though. could you recommend something to search near duplicate images out of 170 million 32x32 images that could support fast insertion/transformation of images of course

**Reply by yhmo** [reply]
Well, I am not an expert in the machine learning area. In [this issue](https://github.com/towhee-io/towhee/issues/480), the Towhee team recommends the Pytorch/Tensorflow to speed up the transformation work. But you need to read the manual of Pytorch/Tensorflow tools and integrate them into your project.
To avoid high dimensional vectors, you can use some [dimensionality reduction](https://towardsdatascience.com/4-ways-to-reduce-dimensionality-of-data-8f82e6565a07) algorithms to convert vectors to low dimensional. You can search "dimensionality reduction" for more information. I just know it is possible to do that but I also don't know how to do it.
